osu.Game/Overlays/Toolbar/DigitalClockDisplay.cs

```diff
@@ -50,13 +66,17 @@ private void load(OsuColour colours)
 
         protected override void UpdateDisplay(DateTimeOffset now)
         {
-            realTime.Text = $"{now:HH:mm:ss}";
+            realTime.Text = format12H ? $"{now:h:mm:ss tt}" : $"{now:HH:mm:ss}";
             gameTime.Text = $"running {new TimeSpan(TimeSpan.TicksPerSecond * (int)(Clock.CurrentTime / 1000)):c}";
         }
 
         private void updateMetrics()
         {
-            Width = showRuntime ? 66 : 45; // Allows for space for game time up to 99 days (in the padding area since this is quite rare).
+            if (format12H)
+                Width = 70;
+            else
+                Width = showRuntime ? 66 : 45; // Allows for space for game time up to 99 days (in the padding area since this is quite rare).
```

Comment:
This layouting is getting really silly now. The digital clock should just autosize, but it's a bit finicky (both clock displays should probably use fixed width for the time part, to avoid the clock weirdly shifting around because of torus digits not being of equal width). Maybe best left for later.

Reply:
It's also too wide. The existing value works quite fine for the AM/PM display as well.

Reply:
Even as-is, this doesn't handle being displayed in a "digital-only" mode very well:

<img width="135" alt="CleanShot 2022-04-07 at 15 49 15@2x" src="https://user-images.githubusercontent.com/22781491/162204447-cd10eac7-6080-4afb-aac2-65d6e2f73691.png">

The way the digital real-time text displays feels quite jarring anyways (especially since that this is updating per-second):

https://user-images.githubusercontent.com/22781491/162205505-23108513-804e-40c9-8126-ed9af3e6fd37.mp4

I'm not sure how good or bad that sounds, but I gave fixed width an attempt, with negative kerning applied, and this is how it looks (on a closer look, the worst it can look is when there are `1` digits on the text):

https://user-images.githubusercontent.com/22781491/162205788-7029e5cc-3ede-42e7-be5c-166df5a80937.mp4

```diff
diff --git a/osu.Game/Overlays/Toolbar/DigitalClockDisplay.cs b/osu.Game/Overlays/Toolbar/DigitalClockDisplay.cs
index 10de35ee50..91b09e78e8 100644
--- a/osu.Game/Overlays/Toolbar/DigitalClockDisplay.cs
+++ b/osu.Game/Overlays/Toolbar/DigitalClockDisplay.cs
@@ -6,6 +6,7 @@
 using osu.Framework.Graphics;
 using osu.Game.Graphics;
 using osu.Game.Graphics.Sprites;
+using osuTK;

 namespace osu.Game.Overlays.Toolbar
 {
@@ -53,9 +54,14 @@ private void load(OsuColour colours)

             InternalChildren = new Drawable[]
             {
-                realTime = new OsuSpriteText(),
+                realTime = new OsuSpriteText
+                {
+                    Font = OsuFont.Default.With(fixedWidth: true),
+                    Spacing = new Vector2(-0.75f),
+                },
                 gameTime = new OsuSpriteText
                 {
+                    X = 2,
                     Y = 14,
                     Colour = colours.PinkLight,
                     Font = OsuFont.Default.With(size: 10, weight: FontWeight.SemiBold),
```

Reply:
https://user-images.githubusercontent.com/22781491/162233191-ab80dad3-8711-4c94-a24d-4fe2d8c3517d.mp4

Also noticed that it starts to look slightly weird when `1` is the first digit after a separator:

<img width="361" alt="CleanShot 2022-04-07 at 18 16 37@2x" src="https://user-images.githubusercontent.com/22781491/162233425-ebca03dc-b040-47df-8aa0-df0efdcb0752.png">

Reply:
that's reason we generally don't use fixed width for numbers. Looks pretty bad there.

Reply:
Fair enough. I'll be fine with this once the layout issue mentioned above is at least fixed.

> Even as-is, this doesn't handle being displayed in a "digital-only" mode very well:
>
> <img width="135" alt="CleanShot 2022-04-07 at 15 49 15@2x" src="https://user-images.githubusercontent.com/22781491/162204447-cd10eac7-6080-4afb-aac2-65d6e2f73691.png">

Reply:
@frenzibyte please dismiss your review if you consider this addressed for now, because it's difficult to tell whether you are still requesting changes right now

Reply:
GitHub's UX is a bit weird on this, but https://github.com/ppy/osu/pull/17670#discussion_r845264465 will need to be solved beforehand.